src/app/Fake.BuildServer.TeamCity/TeamCity.fs

```diff
@@ -216,6 +218,142 @@ module TeamCity =
             (TeamCityWriter.encapsulateSpecialChars name) (TeamCityWriter.encapsulateSpecialChars message) (TeamCityWriter.encapsulateSpecialChars details)
             (TeamCityWriter.encapsulateSpecialChars expected) (TeamCityWriter.encapsulateSpecialChars actual) |> TeamCityWriter.sendStrToTeamCity
 
+    /// TeamCity build parameters
+    /// See [Predefined Build Parameters documentation](https://confluence.jetbrains.com/display/TCD18/Predefined+Build+Parameters) for more information
+    module BuildParameters =
+        open System.IO
+
+        let private get (fileName: string option) =
+            match fileName with
+            | Some fileName when not (isNull fileName) && (File.exists fileName) ->
+                use stream = File.OpenRead(fileName)
+                use reader = new StreamReader(stream)
+
+                reader
+                |> JavaPropertiesFile.parseTextReader
+                |> Seq.choose(function
+                    | JavaPropertiesFile.Comment _ -> None
+                    | JavaPropertiesFile.KeyValue(k, v) -> Some (k,v))
+                |> Map.ofSeq
+            | _ ->
+                Map.empty
+
+        let private systemFile = Environment.environVarOrNone "TEAMCITY_BUILD_PROPERTIES_FILE"
+        let private system = lazy(get (systemFile))
+
+        /// Get all system parameters
+        let getAllSystem () = system.Value
```

Comment:
Somehow I don't like this API `TeamCity.BuildParamters.getAllSystem()`

How about making the module internal and adding

```fsharp
TeamCity.getAllSystemParameters()
```

not sure if that reads better. If you disagree or find a better solution I'm OKish with leaving as is or discussing another alternative.

Reply:
We can, I tried grouping them but it's not mandatory, i'll try to see if I can find good names for all

Reply:
In the end the thing that might read the best is to remove the `tryXXX` and have the getters as static properties

```fsharp
TeamCity.BuildParameters.System |> Map.tryFind "foo"
```

Would it be ok ?